docs/cvl/invariants.md

```diff
@@ -75,6 +78,25 @@ and by {ref}`--parametric_contracts`. `View` and `pure` methods are excluded fro
 invariant checking as by definition they cannot change the state 
 of their contract. 
 
+A strong invariant performs the same checks as a weak invariant - i.e. it will be checked for the constructor
+and for any other method it will be assumed before executing the method (pre-state) and asserted afterward execution of the method (post-state). 
+In addition to these steps, a strong invariant also asserts and assumes the invariant _during_ method
+execution at location that potentially break the invariant. The invariant can be violated if there
+is an unresolved external call that is able to modify the state of the current contract. To verify the strong invariant, for every unresolved external call `c` 
+(a call that will force the prover to havoc storage), a strong invariant will do the following steps:
+
+ 1. Assert that the invariant holds before the `c` - if the invariant does not hold due to some logic of the current method, this will yield a counter example.
+ 2. Assume the invariant holds after the call `c`. The semantics is that the call did not break the invariant.  
+ 3. In the case `c` is a `delegatecall` havoc the current's contact storage and assert the invariant once more. 
+
+The havocing logic in step 3 models the scenario that a `delegatecall` modifies the current contract's storage.
+
+```{note}
+An invariant that is neither declared as `strong` nor `weak` will be treated as a `weak` invariant by default. 
+This behavior can be changed using the flag `--prover_args '-defaultInvariantType strong'`.
```

Comment:
I don't think we should mention this flag:
1. Adds complexity
2. It means we will now have to officially support this prover args syntax
3. If it is a useful option, it should be a flag - flags have better validation, better error messages, and they appear in the help message.
4. The workaround is rather easy (just add `strong` when relevant)

Reply:
I can agree with not mentioning it, (but we should have a list of flags for ourselves)

But the idea was to keep the flag so that we can gather more experience, whether strong or weak is the better default.

Reply:
Regarding flags for internal use:

I bookmarked these two links:
* https://github.com/Certora/EVMVerifier/blob/master/lib/Shared/src/main/kotlin/config/Config.kt
* https://github.com/Certora/EVMVerifier/blob/master/scripts/EVMVerifier/certoraContextAttribute.py